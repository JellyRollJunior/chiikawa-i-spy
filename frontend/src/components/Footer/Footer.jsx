import styles from './Footer.module.css';
import shared from '../../styles/shared.module.css';
import { IconWrapper } from '../IconWrapper/IconWrapper.jsx';

const Footer = () => {
    return (
      <footer className={`${shared.bold} ${shared.marginTopMedium} ${styles.footer}`}>
        <IconWrapper size={32} margin={42} leftIcon={2} rightIcon={2}>
          <div className={styles.iconHolder}>
            <a className={styles.github} href="https://github.com/jellyrolljunior" target="”_blank”"></a>
            <a className={styles.linkedin} href="https://www.linkedin.com/in/jellyrolljunior/" target="”_blank”"></a>
            <a className={styles.instagram} href="https://www.instagram.com/river.flows__" target="”_blank”"></a>
          </div>
        </IconWrapper>
        Created by JellyRollJunior (Brandon Lin)
      </footer>
    );
};

export { Footer }