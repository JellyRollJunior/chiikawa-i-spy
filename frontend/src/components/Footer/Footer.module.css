.footer {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  width: 100%;
  padding: 12px 12px 5px 12px;
  background-color: var(--yellow);

  border-top: 3px solid var(--green);
  border-bottom: 2px solid var(--green);
  border-top-left-radius: 8px;
  border-top-right-radius: 8px;

  text-align: center;
  color: var(--dark-green)
}

.iconHolder {
  display: flex;
  justify-content: center;
  gap: 12px;
  margin-bottom: 3px;
}

.iconLink {
  --link-button-size: 32px;
  display: block;
  width: var(--link-button-size);
  height: var(--link-button-size);
  padding: 0;
  background-size: cover;
}

.github {
  composes: iconLink;
  background-image: url(../../assets/icons/github-60.png);
}

.linkedin {
  composes: iconLink;
  background-image: url(../../assets/icons/linkedin-60.png);
}

.instagram {
  composes: iconLink;
  background-image: url(../../assets/icons/instagram-60.png);
}
