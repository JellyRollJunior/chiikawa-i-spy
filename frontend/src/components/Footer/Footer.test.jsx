import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Footer } from './Footer.jsx';

describe('Footer component', () => {
  it('renders author message and button links', () => {
    const { container } = render(<Footer />);

    expect(container).toMatchSnapshot();``
  });
});
