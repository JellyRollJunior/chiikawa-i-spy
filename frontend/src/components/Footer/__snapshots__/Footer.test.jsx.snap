// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer component > renders author message and button links 1`] = `
<div>
  <div
    class="_forceFooterToBottom_eb428f"
  >
    <div />
    <footer
      class="_bold_5890fc _marginTopMedium_5890fc _footer_eb428f"
    >
      <div
        class="_iconWrapper_5d4a96"
      >
        <img
          alt="Chiikawa character icon"
          src="/src/assets/characterIcons/kurimanju-icon.png"
          style="height: 32px; width: 32px; margin-right: 42px;"
        />
        <div
          class="_iconHolder_eb428f"
        >
          <a
            class="_github_eb428f"
            href="https://github.com/jellyrolljunior"
            target="”_blank”"
          />
          <a
            class="_linkedin_eb428f"
            href="https://www.linkedin.com/in/jellyrolljunior/"
            target="”_blank”"
          />
          <a
            class="_instagram_eb428f"
            href="https://www.instagram.com/river.flows__"
            target="”_blank”"
          />
        </div>
        <img
          alt="Chiikawa character icon"
          src="/src/assets/characterIcons/kurimanju-icon.png"
          style="height: 32px; width: 32px; margin-left: 42px;"
        />
      </div>
      Created by JellyRollJunior (Brandon Lin)
    </footer>
  </div>
</div>
`;
